### PR TITLE
Change `ContainerSelectorGlyph` to use Icon internally

### LINKED ISSUE #9421: Moving icon of GroupBox/Panel is not clear in DemoConsole application
### .NET version

.NET 8.0.100-preview.7.23330.16

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, it doesn't reproduce in .Net 7.0

### Issue description

In [DemoConsole](https://github.com/dotnet/winforms/tree/main/src/System.Windows.Forms/tests/IntegrationTests/DesignSurface/DemoConsole) test app of Winforms repo, the moving icon of GroupBox/Panel is not clear now.

**The result of Main branch for .Net 8.0:**
image

**The result of release/7.0 branch for .Net 7.0:**
image

### Steps to reproduce

Test sample: [DemoConsole](https://github.com/dotnet/winforms/tree/main/src/System.Windows.Forms/tests/IntegrationTests/DesignSurface/DemoConsole) in Winforms repo

### PR DESCRIPTION
**Pull request by elachlan**
Fixes #9421
Caused by #8358

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9432)

### CODE REVIEW

**Comment by elachlan** [comment]
@dreddy-work here is the PR.

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/ContainerSelectorGlyph.cs [hunk 1]
```diff
@@ -44,20 +44,14 @@ public override Rectangle Bounds
     /// </summary>
     public override Cursor? GetHitTest(Point p)
     {
-        if (_glyphBounds.Contains(p) || _relatedBehavior?.OkToMove == true)
-        {
-            return Cursors.SizeAll;
-        }
-
-        return null;
+        return _glyphBounds.Contains(p) || _relatedBehavior?.OkToMove == true ? Cursors.SizeAll : null;
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/ContainerSelectorGlyph.cs:47]: nit: make these expressional body statements.

**Comment by elachlan** [comment]
Should I move the glyph initialization to the constructor? it would simplify the class and avoid creating the icon when paint is called?

**Comment by dreddy-work** [comment]
> Should I move the glyph initialization to the constructor? it would simplify the class and avoid creating the icon when paint is called?

I would move initialization of it to the Paint override instead. that is the only place i see it is being used. If the Container glyph is not selected, It would avoid initialization of Icon.

**Review by dreddy-work — COMMENTED** [review]
LGTM. thank you.

**Review by dreddy-work — APPROVED** [review]

**Comment by dreddy-work** [comment]
@elachlan, in the designer, we made this transparent with following code. Can you capture how it looks with this chnage?
```CS

          static Bitmap LoadBitmap()
          {
              var glyph = new Bitmap(typeof(ContainerSelectorGlyph), "MoverGlyph.bmp");
              glyph.MakeTransparent();

              return glyph;
          }
```

**Comment by elachlan** [comment]
> ```cs
> glyph.MakeTransparent();
> ```

I think I would have to change it back to a bitmap. Load it to an icon and convert it to bitmap. Which might make it not scale well. I'll give it a go.

**Comment by elachlan** [comment]
image
It works.

**Comment by elachlan** [comment]
The change works better with the conversion to bitmap with higher DPIs. I tested 200% on my desktop. The lines are clearer.

**Comment by dreddy-work** [comment]
> The change works better with the conversion to bitmap with higher DPIs. I tested 200% on my desktop. The lines are clearer.
> Bitmap:

We could make Icon load right one with out converting to Bitmap. Did you try transparency?

**Comment by dreddy-work** [comment]
Nevermind, i see code change. Is the screenshots after transparency?

**Comment by elachlan** [comment]
> Nevermind, i see code change. Is the screenshots after transparency?

Yes. Bitmap is after transparency.

**Comment by dreddy-work** [comment]
> Yes. Bitmap is after transparency.

Strange, i do not see difference between transparent and non-transparent here. WinForms designer is using `Cursor.SizeAll` from runtime(windows) one. I do not think we could change that in this. Let me run it by team before merge it.

**Comment by elachlan** [comment]
Could you add waiting on team tag so we don't get lost?

**Comment by dreddy-work** [comment]
With Internal offline discussion, we are taking this as-is for the runtime.

**Comment by dreddy-work** [comment]
@elachlan, before i merge, can you capture a screenshot without `paint` override method in this? Wanted to check if it sets cursor from Windows in those cases ( just like designer)

**Comment by elachlan** [comment]
There is no image on the left hand corner. But the mouse cursor icon changes to the system "move" icon (I am unable to get a screenshot of it).
image

It makes me wonder if we can use a win32 API call to get that icon instead.

Do you still want the `_glyph.MakeTransparent();` call?

**Comment by elachlan** [comment]
Would it be a bad idea to use this?
https://learn.microsoft.com/en-us/windows/win32/menurc/about-cursors
image

**Comment by dreddy-work** [comment]
> Do you still want the `_glyph.MakeTransparent();` call?

Given no difference with transparent, we can remove it.

>Would it be a bad idea to use this?

For this, we may want to try it as a separate PR and validate all the possible scenarios that goes through this call.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**